### ISSUE TITLE
[Feature]: copy a run to another project

### ISSUE THREAD

**Issue by hankyul2** [wandb/wandb#4790]
### Description

copy a run named `my_resnet50` from project `cnn-a` to another project `cnn-b`.

### Suggested Solution

Create a new `Copy` button in this bar.

![image](https://user-images.githubusercontent.com/31476895/212459005-df9345b7-dccb-4161-8299-fc8740950de1.png)

### Alternatives

Use `move` instead of `copy`.

### Additional Context

Thank you.

**Comment by lesliewandb** [comment]
Hi @hankyul2, thank you for writing in about this feature request! I'll create a ticket regarding this for you. But before I do, can you tell me more about why you would want this feature and the use case for it?

**Comment by Nemo1999** [comment]
I really need this feature !
It helps a lot !

**Comment by qgallouedec** [comment]
I would also like to have this functionality. We are building a large reinforcement learning repository. This feature would allow researchers and practitioners to simply copy our runs to compare their algorithms, without having to run them again. 
- Simply copy our runs to compare their algorithms, without having to run them again.
- Contribute more easily to our open benchmark, either with new algorithms/environments or by adding new runs to increase the statistical strength of the results.
See https://wandb.ai/openrlbenchmark/

**Comment by sgaedcke** [comment]
I would also really appreciate such a function! It would help a lot.

**Comment by sheshanshag** [comment]
Same here, would appreciate this functionality! Sometimes, need to compare with an old run in a newer exploration project or compare loss/ other metrics across very different projects.

**Comment by asafarevich** [comment]
Any updates on this?

**Comment by DanielTakeshi** [comment]
Also wondering about this. Thanks!

**Comment by Jibanul** [comment]
Any update about this (very necessary) feature?

**Comment by dchaplinsky** [comment]
I'd subscribe to the updates as well.

**Comment by SHA-T** [comment]
Please, much needed feature!

**Comment by SimiPixel** [comment]
Use case: Sometimes you create a new project, yet a previous run from another project marks the starting point / baseline to which you compare. Then, it would be useful to be able to copy it over into the new project.

**Comment by wobbba** [comment]
I also really want this feature and I have a workaround for the time being:

```python
import wandb

# create dummy
wandb.init(project="PROJECT_NAME")
wandb.finish()

# get runs
api = wandb.Api()
runs = api.runs(path="ENTITY/PROJECT_NAME")

run = runs[1] # the one to copy, edit this selection as needed
dummy = runs[0] # newest, the one we just created

print("duplicating ", run.name)

# We need at least one of these, I didn't check which because it works
dummy_id = dummy._attrs["id"]
dummy_name = dummy._attrs["name"]

for key, value in vars(run).items():
    setattr(dummy, key, value)

dummy._attrs["id"] = dummy_id
dummy._attrs["name"] = dummy_name
dummy._attrs["displayName"] += "_copy" # to avoid confusion

dummy.update()

```

This is not a perfect solution but it does create a new run with the config and summary of the one I'm interested in which is enough for my purposes (I only need it as a comparison/baseline like @SimiPixel mentioned). Hope this helps.

**Comment by Voldemort108X** [comment]
I agree with the above. Being able to get a copy from a previous project is a super helpful feature to keep experiments organized.

**Comment by timoklein** [comment]
For me, this feature would also be very useful to copy baselines between different projects.

**Comment by Voldemort108X** [comment]
Any updates?

**Comment by ankit204** [comment]
Just to support for this feature: this would a meaningful functionality to add to be able to compare one run under multiple different contexts.

**Comment by nickcdryan** [comment]
This seems to do the job. Just add your info at the top. (Adapted from examples [here](https://github.com/wandb/wandb/issues/5726))


```python
import wandb
wandb.login()
api = wandb.Api()

src_entity = "johnsmith"
src_project = "project_1"
src_name = "benchmark_run_1"

dst_entity = "johnsmith"
dst_project = "project_2"

runs = api.runs(f"{src_entity}/{src_project}")

for run in runs:
    if run.name == src_name:
        # Get the run history and files
        history = run.history()
        files = run.files()

        # Create a new run in the destination project
        new_run = wandb.init(project=dst_project, entity=dst_entity, config=run.config, name=run.name,resume="allow")
        
        # Log the history to the new run
        for index, row in history.iterrows():

            # By default copies over the wrong step size for me, so include the run's step_size. Can also enter this manually.
            step_size = history['_step'].values[1]
            
            new_run.log(row.to_dict(), step= index * step_size)

        # Upload the files to the new run
        for file in files:
            file.download(replace=True)
            new_run.save(file.name,policy = "now")

        # Finish the new run
        new_run.finish()
```

**Comment by EricavanderSar** [comment]
Is there an update about this?

**Comment by zhangzw16** [comment]
🙋upvote

**Comment by j1ah0ng** [comment]
upvote

**Comment by DominikaJedynak** [comment]
That would be very helpful

**Comment by jtwild** [comment]
I also want to use this feature!

**Comment by northadventure** [comment]
I really need this

**Comment by diggerk** [comment]
Dear weights & biases, can you comment on the feasibility of this highly requested feature, whether we should expect this to be implemented, or why it can't be done (if that's the case)?

**Comment by Warmund** [comment]
upvote, we need this feature!

**Comment by pengzhenghao-nv** [comment]
we still don't have this today?

**Comment by guosyjlu** [comment]
need this feature!

**Comment by rladbtls1998** [comment]
I need this feature

**Comment by sevmag** [comment]
Also need this feature!

**Comment by wilberquito** [comment]
Dear WandB, update please!

**Comment by rladbtls1998** [comment]
To any people who want to do this, I suggest uploading local wandb files using wandb sync. Not sure if one can download online runs, but if possible, it would be essentially the same as copying while cumbersome.

**Comment by igor-sadalski** [comment]
this feature would make it way easier to compare between different studies! Please implement

**Comment by wilberquito** [comment]
> To any people who want to do this, I suggest uploading local wandb files using wandb sync. Not sure if one can download online runs, but if possible, it would be essentially the same as copying while cumbersome.

This is much more tedious than simply clicking a button, like the one they have for moving runs. As far as I know, when you do that by means of a script, information such as which machine was used for the run gets lost, and some other data isn’t copied correctly.

**Comment by andrey-klochkov-liftoff** [comment]
Maybe a better approach would be to "link" runs from other projects so that metrics could be compared cross-projects for selected runs. This might be conceptually better than having copies of runs, and may be easier to implement (judging by the age of this feature request, it seems to be challenging for wandb to copy runs).

**Comment by leandervaneekelen** [comment]
This feature would be great.

**Comment by bingshuailiu** [comment]
upvote, we need this feature.

**Comment by jws-1** [comment]
Still need this!

**Comment by mutiann** [comment]
Still need this! Or something like a symlink would be OK as well.
